alf/utils/video_recorder.py

```diff
@@ -179,13 +184,57 @@ def capture_frame(self, pred_info=None):
                         "be plotted when rendering videos.")
 
             self.last_frame = frame
+            if encode_frame:
+                if self.ansi_mode:
+                    self._encode_ansi_frame(frame)
+                else:
+                    self._encode_image_frame(frame)
+            else:
+                self._frame_buffer.append(frame)
+
+            assert not self.broken, (
+                "The output file is broken! Check warning messages.")
+
+    def encode_frames_in_buffer(self):
+        # can be extended to encode frames of multiple images
+        for frame in self._frame_buffer:
             if self.ansi_mode:
                 self._encode_ansi_frame(frame)
             else:
                 self._encode_image_frame(frame)
+        # clear frame buffer after encoding all
+        self._frame_buffer = []
 
-            assert not self.broken, (
-                "The output file is broken! Check warning messages.")
+    def encode_frames_in_buffer_with_external(self, set_of_external_frames):
```

Comment:
I feel that the code is bloated by separating this from the original encoding code. I think potentially an external frame or whatever additional info can be first concatenated with the original frame and then call the encoder. BTW, encode_frames_in_buffer has never been called right now.

Reply:
Refactored and moved into VideoRecoder.
Removed encode_frames_in_buffer